tests/test_e2e_80_pathfinder.py

```diff
@@ -0,0 +1,160 @@
+import time
+import pytest
+
+CONTROLLER = '127.0.0.1'
+KYTOS_API = 'http://%s:8181/api/kytos' % CONTROLLER
+
+
+class TestE2EPathfinder:
+    net = None
+
+    def setup_method(self, method):
+        """
+        It is called at the beginning of every class method execution
+        """
+        # Start the controller setting an environment in
+        # which all elements are disabled in a clean setting
+        self.net.start_controller(clean_config=True, enable_all=True)
+        self.net.wait_switches_connect()
+        time.sleep(10)
+
+    @classmethod
+    def setup_class(cls):
+        cls.net = NetworkTest(CONTROLLER, topo_name="multi")
+        cls.net.start()
+        cls.net.restart_kytos_clean()
+        cls.net.wait_switches_connect()
+        time.sleep(5)
+
+    @classmethod
+    def teardown_class(cls):
+        cls.net.stop()
+
+    def restart(self, _clean_config=False, _enable_all=False):
+
+        # Start the controller setting an environment in which the setting is
+        # preserved (persistence) and avoid the default enabling of all elements
+        self.net.start_controller(clean_config=_clean_config, enable_all=_enable_all)
+        self.net.wait_switches_connect()
+
+        # Wait a few seconds to kytos execute LLDP
+        time.sleep(10)
+    
+    def add_topology_metadata(self):
+        """Add topology metadata."""
+        links_metadata = {
+            "74bbc9527a0e309a86c95744042bcf9e3beb52955c942cac5fc735b1cf986f7f": {
+                "link_name": "s1-eth3-s6-eth3",
+                "ownership": "red",
+                "bandwidth": "10",
+                "delay" : 100,
+                "priority": 120,
+            },
+            "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260": {
+                "link_name": "s1-eth2-s2-eth2",
+                "ownership": "blue",
+                "bandwidth": "100",
+                "delay" : 10,
+                "priority": 5,                
+            },
+            "adda3859b963110d584bf6ec3ac85ddea80276001e37edc1c420463a34c80c9e": {
+                "link_name": "s2-eth4-s6-eth4",
+                "ownership": "blue",
+                "bandwidth": "100",
+                "delay" : 10,
+                "priority": 5,                
+            },
+            
+        }
+
+        for link_id, metadata in links_metadata.items():
+            api_url = f"{KYTOS_API}/topology/v3/links/{link_id}/metadata"
+            response = requests.post(
+                api_url,
+                data=json.dumps(metadata),
+                headers={"Content-type": "application/json"},
+            )
+            assert response.status_code == 201, response.text
+        return links_metadata        
+
+    @pytest.mark.parametrize("undesired_link, expected", 
+                            [([], 1),(["74bbc9527a0e309a86c95744042bcf9e3beb52955c942cac5fc735b1cf986f7f"], 8)])
+    def test_10_undesired_link_and_max_path(self, undesired_link, expected):
```

Comment:
For completeness let's also exercise `spf_max_paths` here. Also, let's also include what's the expected path in addition to the expected minimum cost that you've already covered. You can make another request setting `spf_max_paths: 2`, and assert that it has at most 2 paths and which paths are expected too.


Or, alternatively, you can also split this test case into two and cover `spf_max_paths` in a new one.